R/calcOutput.R

```diff
@@ -57,298 +58,322 @@
+#' the data itself and \code{rel} for the relation mapping between countries and regions and
 #' must return the data as magpie object in the spatial resolution as defined in rel.
-#' \item \bold{aggregationArguments} (optional) - List of additional, named arguments to be supplied 
-#' to the aggregation function. In addition to the arguments set here, the function will be 
+#' \item \bold{aggregationArguments} (optional) - List of additional, named arguments to be supplied
+#' to the aggregation function. In addition to the arguments set here, the function will be
 #' supplied with the arguments \code{x}, \code{rel} and if provided/deviating from the default
 #' also \code{weight} and \code{mixed_aggregation}.
 #' }
 #' @author Jan Philipp Dietrich
 #' @seealso \code{\link{setConfig}}, \code{\link{calcTauTotal}},
 #' @examples
-#' 
-#' \dontrun{ 
-#' 
-#' a <- calcOutput(type="TauTotal")
-#' 
+#' \dontrun{
+#'
+#' a <- calcOutput(type = "TauTotal")
 #' }
-#' 
+#'
 #' @importFrom magclass nyears nregions getComment<- getComment getYears clean_magpie write.report2 write.magpie
-#' getCells getYears<- is.magpie dimSums getMetadata updateMetadata
+#' getCells getYears<- is.magpie dimSums
 #' @importFrom utils packageDescription read.csv2 read.csv
 #' @importFrom digest digest
 #' @export
 
-calcOutput <- function(type,aggregate=TRUE,file=NULL,years=NULL,round=NULL,supplementary=FALSE, append=FALSE, na_warning=TRUE, try=FALSE, ...) {
+calcOutput <- function(type, aggregate = TRUE, file = NULL, years = NULL, round = NULL, supplementary = FALSE,
+                       append = FALSE, na_warning = TRUE, try = FALSE, ...) { # nolint
   argumentValues <- c(as.list(environment()), list(...))  # capture arguments for logging
- 
+
   # read region mappings check settings for aggregate
-  if (aggregate!=FALSE) {
+  if (aggregate != FALSE) {
     rel <- list()
-    rel_names <- NULL
-    for(r in c(getConfig("regionmapping"),getConfig("extramappings"))) {
-      rel[[r]] <- toolGetMapping(r, type="regional", activecalc=type)
+    relNames <- NULL
+    for (r in c(getConfig("regionmapping"), getConfig("extramappings"))) {
+      rel[[r]] <- toolGetMapping(r, type = "regional", activecalc = type)
       # rename column names from old to new convention, if necessary
-      if(any(names(rel[[r]])=="CountryCode")) names(rel[[r]])[names(rel[[r]])=="CountryCode"] <- "country"
-      if(any(names(rel[[r]])=="RegionCode")) names(rel[[r]])[names(rel[[r]])=="RegionCode"] <- "region"
-      if(is.null(rel[[r]]$global)) rel[[r]]$global <- "GLO"  # add global column
-      rel_names <- union(rel_names,names(rel[[r]]))
+      if (any(names(rel[[r]]) == "CountryCode")) names(rel[[r]])[names(rel[[r]]) == "CountryCode"] <- "country"
+      if (any(names(rel[[r]]) == "RegionCode")) names(rel[[r]])[names(rel[[r]]) == "RegionCode"] <- "region"
+      if (is.null(rel[[r]]$global)) rel[[r]]$global <- "GLO"  # add global column
```

Comment:
```suggestion
      if (is.null(rel[[r]]$global)) {
        rel[[r]]$global <- "GLO"  # add global column
      }
```
Without this change at first glance it seems like the line below the if statement is executed conditionally, but it's not. Alternatively an extra newline might help, but this is one of the reasons why I prefer "if with {}" almost everywhere